apiserver/DEVELOPMENT.md

```diff
@@ -60,6 +61,59 @@ make build
 make test
 ```
 
+#### Swagger UI updates
+
+To update the swagger ui files deployed with the Kuberay API server, you'll need to:
+
+* Manually run the [hack/update-swagger-ui.bash](hack/update-swagger-ui.bash) script. The script downloads the swagger ui release and copies the files
+located in the [.../third_party/swagger-ui](../third_party/swagger-ui/) directory. It copies the [swagger-initializer.js](../third_party/swagger-ui/swagger-initializer.js)
+to [swagger-initializer.js.backup](../third_party/swagger-ui/swagger-initializer.js.backup).
+* Update the contents of the [swagger-initializer.js](../third_party/swagger-ui/swagger-initializer.js) to set the URLs for for the individual swagger docs. The content of the file is show below:
+
+    ```javascript
+    window.onload = function() {
+    //<editor-fold desc="Changeable Configuration Block">
+
+    // the following lines will be replaced by docker/configurator, when it runs in a docker-container
+    window.ui = SwaggerUIBundle({
+        spec: location.host,
+        urls:  [{"url":"http://"+location.host+"/swagger/serve.swagger.json","name":"RayServe Service"},
+                {"url":"http://"+location.host+"/swagger/error.swagger.json","name":"Errors API"},
+                {"url":"http://"+location.host+"/swagger/job.swagger.json","name":"RayJob Service"},
+                {"url":"http://"+location.host+"/swagger/config.swagger.json","name":"ComputeTemplate Service"},
+                {"url":"http://"+location.host+"/swagger/cluster.swagger.json","name":"Cluster Service"}],
+        dom_id: '#swagger-ui',
+        deepLinking: true,
+        presets: [
+        SwaggerUIBundle.presets.apis,
+        SwaggerUIStandalonePreset
+        ],
+        plugins: [
+        SwaggerUIBundle.plugins.DownloadUrl
+        ],
+        layout: "StandaloneLayout"
+    });
+
+    //</editor-fold>
+    };
+    ```
+
+* Execute `make build-swagger` target to update the contents of the [datafile.go](pkg/swagger/datafile.go) file. This will package the content of the [swagger-ui](../third_party/swagger-ui/) directory for serving by the api server (see [func serveSwaggerUI(mux *http.ServeMux)](https://github.com/ray-project/kuberay/blob/f1067378bc99987f3eba1e5b12b4cc797465336d/apiserver/cmd/main.go#L149) in [main.go](cmd/main.go))
+
+The swagger ui is available at the following URLs:
+
+* [local service -- http://localhost:8888/swagger-ui/](http://localhost:8888/swagger-ui/)
+* [kind cluster -- http://localhost:31888/swagger-ui/](http://localhost:31888/swagger-ui/)
+
+Please note that for the local service to properly serve the swagger content the location of the swagger files would need to be changed in [code](https://github.com/ray-project/kuberay/blob/f1067378bc99987f3eba1e5b12b4cc797465336d/apiserver/cmd/main.go#L138-L142) as follows (lines 138-142 of [main.go](cmd/main.go)):
```

Comment:
I was thinking about this issue, and I wasn't sure if I should include a fix in this PR.  But since you've asked, you have convinced me that we should probably do it. I will push a commit along the lines you have suggested

Reply:
Take a look at the latest commit.